src/checks/#cc4a#prefer_methods.clas.abap

```diff
@@ -0,0 +1,96 @@
+class /cc4a/prefer_methods definition
+  public
+  final
+  create public .
+  public section.
+    interfaces if_ci_atc_check.
+
+    constants:
+      begin of pseudo_comment,
+        avoid_form type string value 'CI_FORM',
+      end of pseudo_comment.
+
+    constants:
+      begin of finding_codes,
+        avoid_form type if_ci_atc_check=>ty_finding_code value 'C_FORM',
+        prefer_methods type if_ci_atc_check=>ty_finding_code value 'C_METHODS',
+      end of finding_codes.
+
+      methods constructor.
+  protected section.
+  private section.
+    data code_provider     type ref to if_ci_atc_source_code_provider.
+    data assistant_factory type ref to cl_ci_atc_assistant_factory.
+    data meta_data         type ref to /cc4a/if_check_meta_data.
+
+    methods analyze_procedure
+      importing procedure       type if_ci_atc_source_code_provider=>ty_procedure
+      returning value(findings) type if_ci_atc_check=>ty_findings.
+
+    methods function_is_rfc_enabled
+      importing io_function_module    type ref to if_xco_function_module
+      returning value(is_rfc_enabled) type abap_bool.
+
+endclass.
+
+class /cc4a/prefer_methods implementation.
+
+  method function_is_rfc_enabled.
+    data(rfc_contract) = io_function_module->content(  )->get_rfc_interface_contract(  ).
+    is_rfc_enabled = xsdbool( rfc_contract is not initial ).
+  endmethod.
+
+  method analyze_procedure.
+    loop at procedure-statements assigning field-symbol(<statement>) where keyword = `FORM` or keyword = `FUNCTION` ##PRIMKEY[KEYWORD].
+      if <statement>-keyword = `FUNCTION`.
+        data(finding_code) = finding_codes-prefer_methods.
+        data(function_module) = xco_cp_abap=>function_module( iv_name = |{ <statement>-tokens[ 2 ]-lexeme }| ).
+        if function_is_rfc_enabled( io_function_module = function_module ).
+          continue.
+        endif.
+      else.
```

Comment:
This works, but in order to understand that this `else` is really an `elseif <statement>-keyword = ``FORM`` ` one has to read the where condition on the loop again. I think this would be clearer if both branches were branches in a case, i.e. 
```abap
case <statement>-keyword.
  when 'FUNCTION'. ...
  when 'FORM'. ...
endcase.
```